Hints.py

```diff
@@ -706,6 +710,36 @@ def buildBossString(reward, color, world):
     return ''
 
 
+def buildBridgeReqsString(world):
+    string = "\x13\x12" # Light Arrow Icon
+    if world.bridge == 'open':
+        string += "The awakened ones will have #already created a bridge# to the castle where the evil dwells."
+    else:
+        item_req_string = getHint('bridge_' + world.bridge, world.clearer_hints).text
+        if world.bridge == 'tokens':
+            item_req_string = str(world.bridge_tokens) + ' ' + item_req_string
+        if '#' not in item_req_string:
+            item_req_string = '#%s#' % item_req_string
+        string += "The awakened ones will await for the Hero to collect %s." % item_req_string
+    return str(GossipText(string, ['Green'], prefix=''))
+
+
+def buildGanonBossKeyString(world):
+    string = "\x13\x74" # Boss Key Icon
+    if world.shuffle_ganon_bosskey == 'remove':
+        string += "And the door of the \x05\x41evil ruler\x05\x40's chamber will be left #unlocked#."
+    else:
+        if 'lacs_' in world.shuffle_ganon_bosskey:
+            item_req_string = getHint(world.shuffle_ganon_bosskey, world.clearer_hints).text
+            if '#' not in item_req_string:
+                item_req_string = '#%s#' % item_req_string
+            bk_location_string = "provided by Zelda once %s are retrieved" % item_req_string
+        else:
+            bk_location_string = getHint('ganonBK_' + world.shuffle_ganon_bosskey, world.clearer_hints).text
+        string += "And the key to the \x05\x41evil\x05\x40 will be %s." % bk_location_string
```

Comment:
How much space do we have? In trying to write this copy in flavor, I might suggest "the evil one's chamber" instead of "the evil" (and "the evil ruler's chamber", honestly) and "the holder of the Triforce of Wisdom" or "the one you call Zelda" in place of "Zelda".

Reply:
We really don't have much space at all, that's the issue. The current options already use as much as we have available in the worst cases (eg. all dungeons or spirit+shadow medallions).

Here are some worst case examples:

![retroarch_2020-09-29_00-22-03](https://user-images.githubusercontent.com/32455037/94492167-f25b2780-01e9-11eb-9724-5bbd0ebb047e.png)

![retroarch_2020-09-29_00-22-14](https://user-images.githubusercontent.com/32455037/94492139-e7a09280-01e9-11eb-94d7-15ae5492bba1.png)

Reply:
How about "the key to the evil king"? I'm not a native speaker but "the key to the evil" sounds very odd to me.

Reply:
"the evil one's key" would work better, then.

Reply:
Is the ampersand (&) available? It only saves a couple characters, but if we're that tight on space...

Reply:
With no further chatter here, I'm going to change "the key to the evil" to "the evil one's key" and nothing else.

Reply:
Looking at it as I merge, I'm also changing remove ganon bosskey to "And the door **to** the evil **one's** chamber" as well.